### PR TITLE
PR for Authentication

### PR DESCRIPTION
**Pull request by SK02K1**
# User authentication added

## Features
- Private route component to validate the authenticated user.
- Login.
- Signup.
- Logout.
- User profile page with controls and logout CTA.
- Toast on success and errors.

### [Deploy Preview](https://deploy-preview-4--focustv.netlify.app/)

Screenshot 2022-03-28 at 10 53 13 PM

Screenshot 2022-03-28 at 10 53 26 PM

Screenshot 2022-03-28 at 10 54 11 PM

Screenshot 2022-03-28 at 10 57 09 PM

Screenshot 2022-03-28 at 10 57 24 PM

Screenshot 2022-03-28 at 11 03 19 PM

### CODE REVIEW

**Review by manismk — APPROVED** [review]
Good folder structure.
Like how you break everything into small pieces.

File: src/pages/index.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
 export { Home } from './Home/Home';
 export { Videos } from './Videos/Videos';
 export { SingleVideo } from './SingleVideo/SingleVideo';
+export { Login } from './Login/Login';
```
Comment (shraddha-amber) [inline src/pages/index.js:4]: suggestion: might perform barrel import

**Review by shraddha-amber — APPROVED** [review]
Very clean and understandable code
Good naming conventions used!

File: src/pages/index.js (hunk 1 above)
Comment (SK02K1) [reply src/pages/index.js:4]: I think you got it wrong, from this file I am exporting all the pages so that other files can use barrel import

**Pull request merged**